pgpy/packet/fields.py

```diff
@@ -345,6 +359,18 @@ def from_signer(self, sig):
         self.s = MPI(seq[1])
 
 
+class EdDSASignature(DSASignature):
+    def from_signer(self, sig):
+        lsig = len(sig)
+        assert lsig % 2 == 0
```

Comment:
if this is a condition worth checking, it's worth raising a real, documentable exception over. Please change this to something like:

```python
if lsig %2 != 0:
    raise PGPError("signature malformed")
```

Reasons for this are:
1. `assert`s are actually skipped when python is run with the `-O` optimize flag, and we can't guarantee nobody is doing that, so we should go out of our way to still provide the same sanity checking for those situations.

2. except where specific, standard exceptions are better (e.g. `KeyError`, `ValueError`), PGPy strives to always raise `PGPError` derived exceptions, to make it more straightforward to isolate recovery or bail-out conditions relating to PGP operations.